packages/js/email-editor/src/components/template-select/select-modal.tsx

```diff
@@ -1,34 +1,83 @@
-// @ts-expect-error No types available for this component
-import { BlockPreview } from '@wordpress/block-editor';
+import { useState, useEffect, memo } from '@wordpress/element';
 import { store as editorStore } from '@wordpress/editor';
 import { dispatch } from '@wordpress/data';
-import {
-	Modal,
-	__experimentalHStack as HStack, // eslint-disable-line
-	Button,
-	Flex,
-	FlexItem,
-} from '@wordpress/components';
+import { Modal, Button, Flex, FlexItem } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-
-import { Async } from './async';
 import { usePreviewTemplates } from '../../hooks';
-import { storeName, TemplatePreview } from '../../store';
+import {
+	EmailEditorPostType,
+	storeName,
+	TemplateCategory,
+	TemplatePreview,
+} from '../../store';
+import { TemplateList } from './template-list';
+import { TemplateCategoriesListSidebar } from './template-categories-list-sidebar';
 
 const BLANK_TEMPLATE = 'email-general';
 
+const TemplateCategories: Array< { name: TemplateCategory; label: string } > = [
```

Comment:
We need to figure out what to do about this Template Category list. 
[WP_Block_Template](https://developer.wordpress.org/reference/classes/wp_block_template/) has an area property we can use to store the template category, but the field and value are not returned from the REST API. 

We would need to add another work-around [here](https://github.com/mailpoet/mailpoet/blob/b1cc53ec5bd0647e347a4ff281791b7e8148bad7/packages/php/email-editor/src/Engine/Templates/class-templates.php#L261-L287) by extending the `wp-template` endpoint. I'm not sure if there's value in doing that now or we should tackle it in this [ticket](https://mailpoet.atlassian.net/browse/MAILPOET-6356).

@costasovo, what do you think?

Reply:
Noted. Thank you for the feedback. 
We will leave it as it is then.